src/main/java/com/exasol/spark/s3/ExasolBatchWrite.java

```diff
@@ -0,0 +1,164 @@
+import software.amazon.awssdk.services.s3.model.S3Object;
+
+/**
+ * An Exasol {@link BatchWrite} implementation.
+ */
+public class ExasolBatchWrite implements BatchWrite {
+    private static final Logger LOGGER = Logger.getLogger(ExasolBatchWrite.class.getName());
+
+    private final ExasolOptions options;
+    private final BatchWrite delegate;
+
+    /**
+     * Creates a new instance of {@link ExasolBatchWrite}.
+     *
+     * @param options  user provided options
+     * @param delegate delegate {@code CSV} batch write
+     */
+    public ExasolBatchWrite(final ExasolOptions options, final Write delegate) {
+        this.options = options;
+        this.delegate = delegate.toBatch();
+    }
+
+    @Override
+    public DataWriterFactory createBatchWriterFactory(final PhysicalWriteInfo info) {
+        return delegate.createBatchWriterFactory(info);
+    }
+
+    @Override
+    public boolean useCommitCoordinator() {
+        return delegate.useCommitCoordinator();
+    }
+
+    @Override
+    public void abort(final WriterCommitMessage[] messages) {
+        LOGGER.info("Running abort stage of the job.");
+        cleanup();
+        delegate.abort(messages);
+    }
+
+    private void cleanup() {
+        LOGGER.info(() -> "Running cleanup process for directory '" + this.options.get(INTERMEDIATE_DATA_PATH) + "'.");
+    }
+
+    @Override
+    public void commit(final WriterCommitMessage[] messages) {
+        LOGGER.info("Committing the file writing stage of the job.");
+        delegate.commit(messages);
+        prepareIntermediateData();
+    }
+
+    private void prepareIntermediateData() {
+        final long start = System.currentTimeMillis();
+        final String table = this.options.getTable();
+        final String query = new S3ImportQueryGenerator(options).generateQuery();
+        try (final Connection connection = new ExasolConnectionFactory(this.options).getConnection();
+                final Statement stmt = connection.createStatement()) {
+            connection.setAutoCommit(false);
+            final int rows = stmt.executeUpdate(query);
+            connection.commit();
+            final long time = System.currentTimeMillis() - start;
+            LOGGER.info(() -> "Imported '" + rows + "' rows into the table '" + table + "' in '" + time + "' millis.");
+        } catch (final SQLException exception) {
+            throw new ExasolConnectionException(ExaError.messageBuilder("E-SEC-24")
+                    .message("Failure running the import {{query}} query.", removeIdentifiedByPart(query))
+                    .mitigation("Please check that connection address, username and password are correct.").toString(),
+                    exception);
+        } finally {
+            cleanup();
+        }
+    }
+
+    private static String removeIdentifiedByPart(final String input) {
+        return Stream.of(input.split("\n")).filter(s -> !s.contains("IDENTIFIED BY")).collect(Collectors.joining("\n"));
```

Comment:
I do not understand. Maybe add a comment to explain, what this does.

Reply:
Fixed

Reply:
For now I renamed this to `removeCredentialsFromQuery`, there is same method in the scan part of the code, later I will try to remove the duplication.